code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -0,0 +1,1376 @@
+	if(linked_moderator)
+		QDEL_NULL(linked_moderator)
+	if(linked_interface)
+		QDEL_NULL(linked_interface)
+	if(corners.len)
+		for(var/corner in corners)
+			QDEL_NULL(corner)
+	QDEL_NULL(radio)
+	QDEL_NULL(soundloop)
+	return..()
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>[src] can be rotated by first opening the panel with a screwdriver and then using a wrench on it.</span>"
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/update_icon()
+	if(panel_open)
+		icon_state = icon_state_open
+	else if(active)
+		icon_state = icon_state_active
+	else
+		icon_state = icon_state_off
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/getNodeConnects()
+	return list(turn(dir, 270), turn(dir, 90))
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/can_be_node(obj/machinery/atmospherics/target)
+	if(anchored)
+		return ..()
+	return FALSE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/attackby(obj/item/I, mob/user, params)
+	if(!fusion_started)
+		if(default_deconstruction_screwdriver(user, icon_state_open, icon_state_off, I))
+			return
+	if(default_change_direction_wrench(user, I))
+		return
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/default_change_direction_wrench(mob/user, obj/item/I)
+	. = ..()
+	if(!.)
+		return
+	if(!anchored)
+		return FALSE
+	var/obj/machinery/atmospherics/node1 = nodes[1]
+	var/obj/machinery/atmospherics/node2 = nodes[2]
+	if(node1)
+		node1.disconnect(src)
+		nodes[1] = null
+		nullifyPipenet(parents[1])
+	if(node2)
+		node2.disconnect(src)
+		nodes[2] = null
+		nullifyPipenet(parents[1])
+
+	SetInitDirections()
+	atmosinit()
+	node1 = nodes[1]
+	if(node1)
+		node1.atmosinit()
+		node1.addMember(src)
+	node2 = nodes[2]
+	if(node2)
+		node2.atmosinit()
+		node2.addMember(src)
+	SSair.add_to_rebuild_queue(src)
+	return TRUE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/check_part_connectivity()
+	. = TRUE
+	if(!anchored || panel_open)
+		return FALSE
+
+	for(var/obj/machinery/hypertorus/object in orange(1,src))
+		if(. == FALSE)
+			break
+
+		if(object.panel_open)
+			. = FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/corner))
+			var/dir = get_dir(src,object)
+			if(dir in GLOB.cardinals)
+				. =  FALSE
+			switch(dir)
+				if(SOUTHEAST)
+					if(object.dir != SOUTH)
+						. = FALSE
+				if(SOUTHWEST)
+					if(object.dir != WEST)
+						. =  FALSE
+				if(NORTHEAST)
+					if(object.dir != EAST)
+						. =  FALSE
+				if(NORTHWEST)
+					if(object.dir != NORTH)
+						. =  FALSE
+			corners |= object
+			continue
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/interface))
+			if(linked_interface && linked_interface != object)
+				linked_interface = null
+				. =  FALSE
+			linked_interface = object
```

Comment:
why use . here?

Reply:
the whole proc is used to check if the 3x3 is all done, and it returns either TRUE (if everything is in place) or FALSE (if something is missing), since it can't use return FALSE i have to use the `.` to have the proc default FALSE if something goes wrong